plugins/Tools.cc

```diff
@@ -377,10 +452,9 @@ void HHAnalyzer::fillTriggerEfficiencies(const Lepton & lep1, const Lepton & lep
         eff_lep2_leg1 = m_hlt_efficiencies.at("IsoMu17leg")->get(p_hlt_lep2)[0];
         eff_lep2_leg2 = m_hlt_efficiencies.at("IsoMu8orIsoTkMu8leg")->get(p_hlt_lep2)[0];
         DZ_filter_eff = DZ_filter_eff_MuMu;
-        // L1 EMTF bug
-        if (getMuonsSector(lep1.p4) == getMuonsSector(lep2.p4) && sameEndcap(lep1.p4, lep2.p4)) {
+        // FIXME L1 EMTF bug
+        if (isCSCSameSectorGen(lep1, lep2))
             DZ_filter_eff *= L1_EMTF_bug_eff_MuMu;
```

Comment:
because the formula is imprecise anyway, it's reverting the full-blown G4 simulation. So it probably can happen that the gen-p4 and reco-p4 are different and this becomes amplified by the back-propagation, leading in turn to different cut efficiencies, leading to data-mc disagreement.

but ok, hopefully muon reconstruction resolutions are small enough to not have such a big effect... If you're not convinced I won't insist :wink: